test/development/app-dir/segment-explorer/segment-explorer.test.ts

```diff
@@ -73,13 +83,16 @@ describe('segment-explorer', () => {
     expect(await getSegmentExplorerContent(browser)).toMatchInlineSnapshot(
       `"app/ [layout.tsx, not-found.js]"`
     )
+    expect(await getSegmentExplorerRoute(browser)).toBe('/404')
   })
 
   it('should show global-error segment', async () => {
     const browser = await next.browser('/runtime-error')
     expect(await getSegmentExplorerContent(browser)).toMatchInlineSnapshot(
       `"app/ [global-error.js]"`
     )
+    // FIXME: handle preserve the url when hitting global-error
```

Comment:
There's a known case when hitting the global error the route will disappear. The whole app state is crashed at that time when hitting the global error boundary. Will handle this in the future